packages/googlecloud/functions/getanswer/main.py

```diff
@@ -62,7 +97,8 @@ def getanswer(request):
         raise ValueError("Unknown content type: {}".format(content_type))
     logging.info("Request parsed")
 
-    answer = answer_query(query, response_type, embeddings)
+    general_embeddings, in_depth_embeddings = create_embeddings()
```

Comment:
This will recompute the embeddings each time the main function is run. This may marginally increase process time if the same function instance is invoked multiple times. An alternative approach may be not to delete `embeddings = OpenAIEmbeddings()` and instead replace it with `embeddings = create_embeddings()`.

I doubt this optimization is very important though.